src/vipersci/carto/bounds.py

```diff
@@ -0,0 +1,86 @@
+"""
+The bounds module is used to calculate a bounding box for a set of 2d spatial data.
+"""
+
+# Copyright 2023, United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Reuse is permitted under the terms of the license.
+# The AUTHORS file and the LICENSE file are at the
+# top level of this library.
+
+import math
+from typing import Tuple
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+def compute_bounds(
+    x_arr: NDArray,
+    y_arr: NDArray,
+) -> Tuple[float, float, float, float]:
+    """Compute cartesian bounds of input coordinates
+
+    Args:
+        x_arr: set of x locations
+        y_arr: set of y locations
+
+    Returns:
+        a tuple (left, bottom, right, top) describing the bounds of the input data
+    """
+    return (np.amin(x_arr), np.amin(y_arr), np.amax(x_arr), np.amax(y_arr))
+
+
+def pad_grid_align_bounds(
+    left: float,
+    bottom: float,
+    right: float,
+    top: float,
+    ground_sample_distance: float,
+    padding: int = 0,
+) -> Tuple[float, float, float, float]:
+    """Given bounds, add a padding and align to a grid of size of
+    ground_sample_distance.  That is, the output bounds will be integer multiples of
+    ground_sample_distance.
+
+    Args:
+        left: xmin of input bounds
+        bottom: ymin of input bounds
+        right: xmax of input bounds
+        top: ymax of input bounds
+        ground_sample_distance: Spatial resolution of grid
+        padding: Pixels (multiples of ground_sample_distance) of padding to add on to bounds
+    Returns:
+        a tuple (left, bottom, right, top) describing padded and aligned bounds
+    """
+
+    # Convert from pixels of padding to a buffer in x/y distance units.
+    buffer = padding * ground_sample_distance
+    # Make sure that bounds are snapped into a grid centered on the
+    # origin with a *ground_sample_distance* step.
+    left = math.floor((left - buffer) / ground_sample_distance) * ground_sample_distance
```

Comment:
One last fiddly request.  Now that these are laid out, this looks like a repeated pattern, and that's always an indication that a function is probably better (if you find yourself cutting and pasting, that's usually an opportunity).  So I'm going to suggest the following private function for this module (hence the starting with an underbar):
```
def _grid_snap(operator, bufferedvalue, gsd):
    return getattr(math, operator)(bufferedvalue / gsd) * gsd
```
I suppose it could even be an internal function to *pad_grid_align_bounds()*.

And then you can write these lines like:
```
left = _grid_snap("floor", left - buffer, ground_sample_distance)
right = _grid_snap("ceil", right + buffer, ground_sample_distance)
```
You don't gain a lot, but it tidies it up a miniscule amount.